### DISCUSSION TITLE
How to populate version of current project from cmake or source file - into vcpkg.json?

### DISCUSSION THREAD

**Discussion by sairus7** [microsoft/vcpkg#25693] [Q&A]
Or any other way to sync the same version between CMake, source file and vcpkg?

For now I just write version from CMakeLists.txt into a header file:
```cmake
set(MY_PROJECT_VERSION "1.0.0")
configure_file(version.h.in version.h @ONLY)
include_directories(${CMAKE_CURRENT_BINARY_DIR})
```

**Comment by Osyotr** [comment]
You don't really need to have meaningful values in a manifest file. There is even a PR that will make it possible to omit port name and version in manifest: https://github.com/microsoft/vcpkg-tool/pull/605
As for cmake + header file, I would go the other way around: read `version.h` from source folder and set version in cmake accordingly.
